go.mod

```diff
@@ -2,19 +2,21 @@ module sigs.k8s.io/jobset
 
 go 1.23.0
 
+toolchain go1.24.1
```

Comment:
Should we update Go to 1.24 to avoid this toolchain ?

Reply:
cc @tenzen-y @astefanutti

Reply:
I remember that @kannon92 wants to keep using 1.23 for some reason.

Reply:
It was mostly because 0.32 is still really 1.23.

I was thinking we can upgrade to 1.24 once the k8s dependencies require 1.24.

If someone wants to upgrade feel free!